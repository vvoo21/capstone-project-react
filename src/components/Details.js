import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import fetchCryptoDetail from '../redux/cryptoDetailsApi';

const Details = () => {
  const state = useSelector((state) => state.details);
  const dispatch = useDispatch();

  const { id } = useParams();

  useEffect(() => {
    dispatch(fetchCryptoDetail(id));
  }, [dispatch]);

  console.log(state);

  return (
    <div>
      {state.map((cryptoDetail) => (
        <div key={cryptoDetail.FROMSYMBOL}>
          <div>
            <img src={`https://www.cryptocompare.com/${cryptoDetail.IMAGEURL}`} alt="crypto-img" />
            <h2>{cryptoDetail.FROMSYMBOL}</h2>
          </div>
          <ul>
            <li>
              <p>
                The price is:
                <span>{cryptoDetail.PRICE}</span>
              </p>
            </li>
            <li>
              <p>
                The highest price of the day:
                <span>{cryptoDetail.HIGHDAY}</span>
              </p>
            </li>
            <li>
              <p>
                The lowest price of the day:
                <span>{cryptoDetail.LOWDAY}</span>
              </p>
            </li>
            <li>
              <p>
                Variation last 24 hours:
                <span>{cryptoDetail.CHANGEPCT24HOUR}</span>
              </p>
            </li>
            <li>
              <p>
                Last update:
                <span>{cryptoDetail.LASTUPDATE}</span>
              </p>
            </li>
          </ul>
        </div>
      ))}
    </div>
  );
};

export default Details;
