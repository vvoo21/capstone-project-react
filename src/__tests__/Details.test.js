import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import Details from '../components/Details';
import cryptoDetailsSlice from '../redux/cryptoDetails';

const createTestStore = () => {
  const store = configureStore({
    reducer: {
      details: cryptoDetailsSlice,
    },
  });

  return store;
};
export default createTestStore;

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    component = renderer.create(
      <Provider store={store}>
        <Details />
      </Provider>,
    ).toJSON();
  });

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});