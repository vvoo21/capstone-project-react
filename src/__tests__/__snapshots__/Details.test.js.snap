// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div
  className="details-container"
>
  <div>
    <img
      height="150"
      src="https://www.cryptocompare.com/undefined"
      width="150"
    />
    <h2
      className="text-symbol"
    >
      Symbol:  
    </h2>
  </div>
  <ul
    className="details-list"
  >
    <li>
      <p
        className="info-text"
      >
        coin information
      </p>
    </li>
    <li>
      <p
        className="info-text"
      >
        The price is:  
      </p>
    </li>
    <li>
      <p
        className="info-text"
      >
        The highest price of the day:  
      </p>
    </li>
    <li>
      <p
        className="info-text"
      >
        The lowest price of the day:  
      </p>
    </li>
    <li>
      <p
        className="info-text"
      >
        Variation last 24 hours:  
      </p>
    </li>
    <li>
      <p
        className="info-text"
      >
        Last update:  
      </p>
    </li>
  </ul>
</div>
`;
